python3Packages.simpleldap: remove

Ma27 opened the pull request:
###### Motivation for this change

Fairly outdated and incompatible with Python 3, see the following links
for further reference:

* https://github.com/gdub/python-simpleldap/blob/0.8/setup.py#L12-L14
* #43782 (comment)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Ma27 committed:
python3Packages.simpleldap: remove

Fairly outdated and incompatible with Python 3, see the following links
for further reference:

* https://github.com/gdub/python-simpleldap/blob/0.8/setup.py#L12-L14
* https://github.com/NixOS/nixpkgs/pull/43782#issuecomment-406186881

dotlambda commented:
Judging from https://github.com/gdub/python-simpleldap/blob/0.8/setup.py#L12-L14, we should set `disabled = isPy3k`.

dotlambda commented on pkgs/development/python-modules/simpleldap/default.nix:
```diff
@@ -0,0 +1,31 @@
+  pname = "simpleldap";
+  version = "0.8";
+
+  src = fetchFromGitHub {
```
What's the reason you switched to `fetchFromGitHub`? Please add an inline comment.

dotlambda commented:
I think we should just remove this package: It hasn't been updated since 2014 and its only dependent, inginious, is marked as broken since https://github.com/NixOS/nixpkgs/commit/611116823a13930caf2bf0402300b9b523efb92b. It should either be updated to the most recent version, which replaces `simpleldap` by `ldap3` or completely removed as @layus suggested in https://github.com/NixOS/nixpkgs/pull/31975#issuecomment-346647857.

Ma27 committed:
inginious: remove

Fallout of 9db7f15ea3ce57eee25458daf87cce7a04ea98b2 which removed
simpleldap.

@layus suggests to remove:

* https://github.com/NixOS/nixpkgs/pull/43782#issuecomment-406186881
* https://github.com/NixOS/nixpkgs/pull/31975#issuecomment-346647857

Ma27 commented:
@dotlambda thanks for elaborating, I dropped both of these packages, a broken one is obviiously not in use and `simpleldap` is way too outdated which is why I should use a more recent alternative :sweat_smile:

dotlambda commented:
> I dropped both of these packages

Please adjust the commit message of the first commit accordingly.

Ma27 commented:
argh, good catch, sorry :/
